packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceModal.tsx

```diff
@@ -0,0 +1,483 @@
+import { Classes, Dialog, Spinner } from '@blueprintjs/core';
+import {
+    ItemPredicate,
+    ItemRenderer,
+    MultiSelect2,
+    Select2,
+} from '@blueprintjs/select';
+import { OrganizationMemberProfile, Space } from '@lightdash/common';
+import { FC, useCallback, useMemo, useState } from 'react';
+import { useOrganizationUsers } from '../../../hooks/useOrganizationUsers';
+import { useProject } from '../../../hooks/useProject';
+import { useProjectAccess } from '../../../hooks/useProjectAccess';
+import {
+    AccessDescription,
+    AccessName,
+    AccessRole,
+    AccessSelectSubtitle,
+    AccessSelectTitle,
+    AccessWrapper,
+    AddUsersWrapper,
+    ChangeAccessButton,
+    DialogFooter,
+    FlexWrapper,
+    MemberAccess,
+    OpenShareModal,
+    ShareButton,
+    ShareTag,
+    UserName,
+    UserRole,
+    UserTag,
+    YouLabel,
+} from './ShareSpaceModal.style';
+
+import { MenuItem2 } from '@blueprintjs/popover2';
+import {
+    useAddSpaceShareMutation,
+    useDeleteSpaceShareMutation,
+    useUpdateMutation,
+} from '../../../hooks/useSpaces';
+import { useApp } from '../../../providers/AppProvider';
+export interface ShareSpaceProps {
+    space: Space;
+    projectUuid: string;
+}
+
+export interface AccessOption {
+    title: string;
+    description?: string;
+    selectDescription: string;
+    value: string;
+}
+const StyledSpinner = () => <Spinner size={16} style={{ margin: 12 }} />;
+
+const capitalize = (word: string): string =>
+    word ? `${word.charAt(0).toUpperCase()}${word.slice(1).toLowerCase()}` : '';
+
+const enum SpaceAccessType {
+    PRIVATE = 'private',
+    PUBLIC = 'public',
+}
+const SpaceAccessOptions: AccessOption[] = [
+    {
+        title: 'Restricted access',
+        description: 'Only invited members can access',
+        selectDescription: 'Only invited members can access',
+        value: SpaceAccessType.PRIVATE,
+    },
+    {
+        title: 'Full access',
+        description: 'All project members can access',
+        selectDescription:
+            'All project members can access with their project permissions',
+        value: SpaceAccessType.PUBLIC,
+    },
+];
+
+const enum UserAccessAction {
+    KEEP = 'keep',
+    DELETE = 'delete',
+}
+
+const UserAccessOptions: AccessOption[] = [
+    {
+        title: 'viewer',
+        selectDescription: `This permission has been inherited from user's project access`,
+        value: UserAccessAction.KEEP,
+    },
+    {
+        title: 'No access',
+        selectDescription: `Remove user's access`,
+        value: UserAccessAction.DELETE,
+    },
+];
+
+const renderAccess: ItemRenderer<AccessOption> = (
+    access,
+    { handleClick, handleFocus, modifiers, query },
+) => {
+    if (!modifiers.matchesPredicate) {
+        return null;
+    }
+    return (
+        <MenuItem2
+            multiline={true}
```

Comment:
it will be better not to use this and have a 1 row, unwrapped text
<img width="880" alt="CleanShot 2022-11-16 at 13 18 54@2x" src="https://user-images.githubusercontent.com/962095/202140537-bd3a042e-9e97-4033-aa23-bfaea51a8a4a.png">

Reply:
If I remove multiline.. 

![image](https://user-images.githubusercontent.com/1983672/202177602-148f9aea-1bb2-43a5-894d-2f12c7bce83b.png)